Added CI with RISC-V RVV1.0 Hardware

alitariq4589 opened the pull request:
I have added the RISC-V CI for running RISC-V RVV 1.0 hardware

alitariq4589 committed:
Changed the CI file to hw

alitariq4589 commented:
Still refactoring.

alitariq4589 committed:
Changed the CI file to hw

alitariq4589 committed:
Added to sudoers for apt

alitariq4589 committed:
Removed the clone command and used checkout

alitariq4589 committed:
Added libcurl

alitariq4589 committed:
Added gcc-14

alitariq4589 committed:
Checking gcc --version

alitariq4589 committed:
added gcc-14 symlink

alitariq4589 committed:
added CC and C++ variables

alitariq4589 committed:
Added the gguf weight

alitariq4589 committed:
Changed the weights path

alitariq4589 committed:
Added system specification

alitariq4589 commented:
I have updated the CI file in the repository to run on real RISC-V hardware with RVV 1.0, instead of using QEMU.

A working CI log on [RISC-V Jupiter board ](https://milkv.io/jupiter)can be seen [here](https://dash.cloud-v.co/view/all/job/llama.cpp%20--%20fork%20for%20testing/15/console).

@ggerganov Your credentials were already added, you just have to add a webhook for the project (unless the credentials have expired). Webhook was previously disabled somehow. It would be great if you could add the webhook again.

You will just have to add the following details to the webhook settings of the repository. 

![image](https://github.com/user-attachments/assets/7725f0d1-10c1-487c-8742-ae10b4b44ea0)


In the events, select Pull requests and Pushes.

Once you have added this, I will try to test by creating a PR (which I will close later if the build is triggering).

alitariq4589 committed:
Removed white spaces

alitariq4589 commented:
Reference PR: #3160

alitariq4589 commented:
I have now added a github CI instead of Jenkins for RISC-V vector. You will just have to go to this [link](https://cloud-v.co/github-riscv-runner) and add the github runner token with the repository URL. The compute machine `Banana Pi BPI-F3` is a physical RISC-V board which has RVV1.0, so if you choose it, CI will run on RVV1.0. Plus, I will remove the 90-day unregistration limit so the project stays registered.

This does not require Jenkinsfile to be merged; github CI will run using github actions/workflows.

alitariq4589 commented:
@ggerganov 

In your repository, navigate to `Settings -> Actions -> Runners -> New self-hosted runner`, and retrieve the token at the end and the repository link. Once you register, I will remove the 90-day limit from this repository, allowing you permanent CI access, as it is an active repository.

This may also need a workflow file for running on RISC-V. I will change the already added script to be github native workflow file instead of a Jenkins pipeline.

<img width="1874" height="875" alt="Selection_002" src="https://github.com/user-attachments/assets/84626b26-4473-481c-a0ac-1e29b4c147e5" />

  
  
On the next page...

  
  
<img width="1112" height="424" alt="Selection_003" src="https://github.com/user-attachments/assets/f520f6fd-ec46-478b-8d34-0d6b503b31d8" />

ggerganov commented:
Thanks. I have completed the registration of the runner in cloud-v. Anything else needed on my side?

alitariq4589 commented:
@ggerganov 

Thanks. I can see it is registered in the database. There is nothing needed from your side at the moment. I will delete the Jenkinsfile, which was previously added for Cloud-V for QEMU, and then will add a GitHub Actions file for compilation and running on RISC-V. 

Once that is done, this PR will have to be merged.

alitariq4589 commented:
@ggerganov 

I think you may have added the expired token. I am getting an authentication error inside GitHub Actions.


```
--------------------------------------------------------------------------------
|        ____ _ _   _   _       _          _        _   _                      |
|       / ___(_) |_| | | |_   _| |__      / \   ___| |_(_) ___  _ __  ___      |
|      | |  _| | __| |_| | | | | '_ \    / _ \ / __| __| |/ _ \| '_ \/ __|     |
|      | |_| | | |_|  _  | |_| | |_) |  / ___ \ (__| |_| | (_) | | | \__ \     |
|       \____|_|\__|_| |_|\__,_|_.__/  /_/   \_\___|\__|_|\___/|_| |_|___/     |
|                                                                              |
|                       Self-hosted runner registration                        |
|                                                                              |
--------------------------------------------------------------------------------

# Authentication

Http response code: NotFound from 'POST https://api.github.com/actions/runner-registration' (Request Id: AB16:21A34E:7A84D:860C3:689B253A)
{"message":"Not Found","documentation_url":"https://docs.github.com/rest","status":"404"}
Response status code does not indicate success: 404 (Not Found).

```


I am removing this registration. Can you please add it again?

ggerganov commented:
Just to make sure - do I need to actually create the runner in the Github settings, or just to copy the token from the page?

alitariq4589 commented:
From what I can see in the curl output, there is some issue with how the token is generated.

```
HTTP/2 401 
date: Tue, 12 Aug 2025 11:54:37 GMT
content-type: application/json; charset=utf-8
content-length: 109
x-github-media-type: github.v3; format=json
x-ratelimit-limit: 60
x-ratelimit-remaining: 57
x-ratelimit-reset: 1755003119
x-ratelimit-used: 3
x-ratelimit-resource: core
access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
access-control-allow-origin: *
strict-transport-security: max-age=31536000; includeSubdomains; preload
x-frame-options: deny
x-content-type-options: nosniff
x-xss-protection: 0
referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
content-security-policy: default-src 'none'
vary: Accept-Encoding, Accept, X-Requested-With
server: github.com
x-github-request-id: B8C4:1B362B:D62C2:EAF39:689B2B7C

{
  "message": "Bad credentials",
  "documentation_url": "https://docs.github.com/rest",
  "status": "401"
}
```

According to github documentation for rest api:

>[Failed login limit](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28#failed-login-limit)
If you try to use a REST API endpoint without a token or with a token that has insufficient permissions, you will receive a 404 Not Found or 403 Forbidden response. Authenticating with invalid credentials will initially return a 401 Unauthorized response.
>
>After detecting several requests with invalid credentials within a short period, the API will temporarily reject all authentication attempts for that user (including ones with valid credentials) with a 403 Forbidden response. For more information, see [Rate limits for the REST API](https://docs.github.com/en/rest/overview/rate-limits-for-the-rest-api).

alitariq4589 commented:
That explains the issue. I am outside right now. I will message you after removing the token after a while.

(Hopefully this will work out this time. And sorry for prolonged setup issue 😅)

ggerganov commented:
ETA 30 min (afk atm)

ggerganov commented:
No problem and thank you for providing hardware for the CI - appreciated!

akifejaz committed:
Merge branch 'master' into alitariq4589/rvv_ci

akifejaz committed:
ci: Replace Jenkins riscv native build Cloud-V pipeline with GitHub Actions workflow

Removed the legacy .devops/cloud-v-pipeline Jenkins CI configuration and introduced .github/workflows/build-riscv-native.yml for native RISC-V builds using GitHub Actions.

akifejaz committed:
Merge pull request #2 from alitariq4589/ae/rvv-native-build

Replace Jenkins Cloud-V pipeline with GitHub Actions RISC-V native build

akifejaz committed:
removed trailing whitespaces

ixgbe commented:
> I have updated the CI file in the repository to run on real RISC-V hardware with RVV 1.0, instead of using QEMU.
> 
> A working CI log on [RISC-V Jupiter board ](https://milkv.io/jupiter)can be seen [here](https://dash.cloud-v.co/view/all/job/llama.cpp%20--%20fork%20for%20testing/15/console).
> 
> @ggerganov Your credentials were already added, you just have to add a webhook for the project (unless the credentials have expired). Webhook was previously disabled somehow. It would be great if you could add the webhook again.
> 
> You will just have to add the following details to the webhook settings of the repository.
> 
> ![image](https://private-user-images.githubusercontent.com/99069972/460305449-7725f0d1-10c1-487c-8742-ae10b4b44ea0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY3MTAyMzMsIm5iZiI6MTc1NjcwOTkzMywicGF0aCI6Ii85OTA2OTk3Mi80NjAzMDU0NDktNzcyNWYwZDEtMTBjMS00ODdjLTg3NDItYWUxMGI0YjQ0ZWEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTAxVDA2NTg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljMDA0ZDEzYWUyOTk0OWJiZmU1ZmU3NzVlZjZlODMzOGQxZGY4YzE5NWZiNzBhNmRhNDM2OTI5ZDZkNzNhMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UZTxM1zvUJhRWyW1NnxV3h0p_muTokQI94rBv8LfroM)
> 
> In the events, select Pull requests and Pushes.
> 
> Once you have added this, I will try to test by creating a PR (which I will close later if the build is triggering).

Hi , I have a question about the webhook configuration in this project.

Is the **Payload URL** required for CI builds to work?  
In our current setup, we're using self-hosted runners on RISC-V machines, and I'm trying to understand whether the webhook (Payload URL) is actually being used in this project.

Specifically:
1. Is the Payload URL necessary for GitHub Actions workflows to trigger?
2. Are we currently using any webhook payloads in our CI logic (e.g., to trigger builds, pass data, etc.)?

I noticed that we have a webhook configured, but I'm not sure if it's actively used, especially since our workflows are triggered by `pull_request` and `workflow_dispatch`.

Any clarification would be greatly appreciated!

Thanks in advance!
